smt/surrogate_models/neural_net/optimizer.py

```diff
@@ -0,0 +1,351 @@
+            self._design_history.append(self._current_design.copy())
+
+            if is_print:
+                print("iteration = {:d}, cost = {:6.3f}".format(i, float(self._current_cost)))
+
+            # Absolute convergence criterion
+            if i > 1:
+                dF1 = abs(self._cost_history[-1] - self._cost_history[-2])
+                if dF1 < epsilon_absolute * self._cost_history[0]:
+                    N1 += 1
+                else:
+                    N1 = 0
+                if N1 > N1_max:
+                    converged = True
+                    if is_print:
+                        print("Absolute stopping criterion satisfied")
+
+                # Relative convergence criterion
+                dF2 = abs(self._cost_history[-1] - self._cost_history[-2]) / max(abs(self._cost_history[-1]), 1e-6)
+                if dF2 < epsilon_relative:
+                    N2 += 1
+                else:
+                    N2 = 0
+                if N2 > N2_max:
+                    converged = True
+                    if is_print:
+                        print("Relative stopping criterion satisfied")
+
+                # Maximum iteration convergence criterion
+                if i == max_iter:
+                    if is_print:
+                        print("Maximum optimizer iterations reached")
+
+                if converged:
+                    break
+
+        self._optimum_design = self._current_design.copy()
+
+        return self.optimum
+
+
+class GD(Optimizer):
+
+    def _update_current_design(self, learning_rate=0.05):
+        """Gradient descent update"""
+        for key in self._previous_design.keys():
+            self._current_design[key] = self._previous_design[key] - learning_rate * self._search_direction[key]
+
+
+class Adam(Optimizer):
+
+    def __init__(self, **kwargs):
+        super(Adam, self).__init__(**kwargs)
+        self.v = None
+        self.s = None
+
+    def _update_current_design(self, learning_rate=0.05, beta_1=0.9, beta_2=0.99):
+        """Adam update"""
+        self.beta_1 = beta_1
+        self.beta_2 = beta_2
+        t = self._current_iteration + 1
+
+        if self.v is None:
+            self.v = {key: np.zeros(value.shape) for key, value in self._current_design.items()}
+        if self.s is None:
+            self.s = {key: np.zeros(value.shape) for key, value in self._current_design.items()}
+
+        for key in self._current_design.keys():
+            self.v[key] = self.beta_1 * self.v[key] + (1. - beta_1) * self._search_direction[key]
+            self.s[key] = self.beta_2 * self.s[key] + (1. - beta_2) * np.square(self._search_direction[key])
+            v_corrected = self.v[key] / (1. - self.beta_1 ** t)
+            s_corrected = self.s[key] / (1. - self.beta_2 ** t)
+            self._current_design[key] = self._previous_design[key] - learning_rate * v_corrected / (np.sqrt(s_corrected) + EPS)
+
+
+def test_optimizer(use_adam=True):
```

Comment:
Should be moved in test file

Reply:
Renamed and #pragma no cover. It was mostly used for development and doesn't need to be tested again.